contracts/lib/TokenDebtTracker.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.13;
+
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import { SafeERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+
+/**
+ * @title TokenDebtTracker
+ * @author Venus
+ * @notice TokenDebtTracker is an abstract contract that handles transfers _out_ of the inheriting contract.
+ * If there is an error transferring out (due to any reason, e.g. the token contract restricted the user from
+ * receiving incoming transfers), the amount is recorded as a debt that can be claimed later.
+ * @dev Note that the inheriting contract keeps some amount of users' tokens on its balance, so be careful when
+ * using balanceOf(address(this))!
+ */
+abstract contract TokenDebtTracker is Initializable {
```

Comment:
I expected to see a library here because it's in the folder `lib` :)

Reply:
I treat it as "a library of smart contracts", just like [oz](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/02ea01765a9964541dd9cdcffc4a7f8b403c2ff6/README.md). But we can use some different convention, e.g. `utils` for (potentially abstract) contracts and `lib` for libraries.